builtin/logical/pki/cert_util.go

```diff
@@ -97,24 +133,30 @@ func fetchCertBySerial(req *logical.Request, prefix, serial string) (*logical.St
 // Given a set of requested names for a certificate, verifies that all of them
 // match the various toggles set in the role for controlling issuance.
 // If one does not pass, it is returned in the string argument.
-func validateCommonNames(req *logical.Request, commonNames []string, role *roleEntry) (string, error) {
+func validateNames(req *logical.Request, names []string, role *roleEntry) (string, error) {
```

Comment:
I sort of gave up trying to understand this function, it has too many complex conditions going on. It needs to be refactored and tested.